.github/workflows/build_and_test.yml

```diff
@@ -15,684 +15,725 @@ permissions:
   contents: read
 
 jobs:
-  pre-commit:
+  doxygen:
     runs-on: ubuntu-latest
-    container: seladb/alpine317
+    container: seladb/ubuntu1804:latest
```

Comment:
@seladb why ubuntu18? we should use the lastest LTS.

Reply:
We need doxygen to be installed. We installed it in all of our Ubuntu images: https://github.com/seladb/PcapPlusPlus-DockerImages/blob/master/Dockerfile-ubuntu2204

I updated the image to Ubuntu 22.04